### ISSUE TITLE
anime.js overrides static css transforms?

### ISSUE THREAD

**Issue by plagasul** [juliangarnier/anime#415]
Adding a static css transform on a `svg path` element and then animating the same element using anime.js overrides the original css transform. 

In example if I attempt a` transform: rotate(45deg)` on the static css, and then I run anime.js like this:

```
	anime({
		targets: "svg path#uno",
		rotate: "-360deg",
		duration: 8000,
		loop: true,
		easing: "linear"
	})
```
... the element starts rotating from 0deg not 45deg as expected.

Visually, on load, I get a quick jump from the 45º rotated state to the 0º rotated state and then the animation starts.

The same happens with other css transforms, like translate.

If I cannot transform the svg path with regular CSS BEFORE the animation... can I do it within anime itself? 

I mean:

```
on load
inmediately transform(rotate,translate...) the svg path ONCE
THEN rotate -360deg during 8000ms ON LOOP
```

I could just transform the svg before pasting it on my code, but I would like to randomize this transform on load.

Thank you

**Comment by arbel** [comment]
I have the same issue with a scale, I want the object to start scaled at 0.1 and then animate to scale 1.

**Comment by plagasul** [comment]
A workaround I am using is using anime's timeline to do the initial transform at 0 duration, then the rest with an offset just in case:

var timeline = anime.timeline();

timeline
	.add({
		targets: '#target',
		rotate: 120,
		duration: 0	,
		loop: false,
		easing: "easeInQuad",
		offset: 0 
	})
	.add({
		targets: '#target',
		rotate: 360,
		duration: 8000,
		loop: true,
		easing: "linear",
		offset: 10
	})	


The only problem is that the element to transform (inline svg) loads before the first transformation, so first it flashes without any transformation, then the timeline starts. In my case I solved it with a fade in that starts after the first transform, so there is no flash. But I was going to do the fadein anyway.

**Comment by arbel** [comment]
but what happens when the animation completes.....I had some wiered issues there.....seems like a very odd feature/bug

**Comment by arbel** [comment]
`var basicTimeline = anime.timeline({autoplay: true,  loop:false});

		basicTimeline
		
		 
		 .add({
		    targets: '.course .tablet',
		    scale: 0.5,		   
		    easing: 'easeOutExpo',
		    duration:0,
		    offset:0
		  })
		  .add({
		    targets: '.course .tablet',
		    scale: 1,
		    opacity:1,
		    easing: 'easeOutExpo',
		    duration:3000,
		    offset:400
		  });`

this is my code, but after the animation is done, the "tablet" resets to scale:0.5

**Comment by arbel** [comment]
any advice on this?

**Comment by plagasul** [comment]
@arbel can you confirm that works for you ?

**Comment by Akhu** [comment]
Got the same issue here. My workaround is to define initial value with js at first (not very handy). I will try your solution @plagasul Thanks

**Comment by Akhu** [comment]
This is working for me ! Thanks again @plagasul 
Quick sample :
```js
basicTimeline.add({ 
        targets: '#id',
        opacity: [0,1], 
        rotateX: ['40deg', '0deg'], //Setting initial and target values for the transform
        translateY: ['-10px','0px'],
        easing: easingCustom,
        duration: 500
});
```

**Comment by plagasul** [comment]
Great @Akhu  :) 

The only problem I see with this technique is that it may lead to a "flash" of the content not set to initial value until the js is loaded, which may be very visible if the js takes too long to load.

If I am right about this, a possible solution would be to establish the initial value on css, grab it from there on the js and use it as the initial value in anime.

**Comment by BrianAA** [comment]
Agreed with this issue I am coming across the same. One workaround is set the opacity to 0 in the CSS class and then animated into view making sure you set the opacity 1.  You could offset the animation in the timeline so it's in sync with the rest of your animation.

**Comment by juliangarnier** [comment]
Anime.js can't get initial targets CSS properties values defined in a CSS file, they have to be inlined.
You can now use `anime.set()` to set transforms values instantly before running your animation : https://animejs.com/documentation/#set

**Comment by janstieler** [comment]
Hi,
I think I have a similar problem but my CSS is inlined with a style-element through jSBin.
I have a svg-logo and I have to animate some of the logo elements. I reordered it with the initial CSS different from the svg attributes.

I have there a timeline ([Line 26](https://jsbin.com/firetuh/edit?html,css,js,output#J:L26)) which is slices with keyframes where I set the transform for the animation of the element `#point01`. This Element switchs back to the old startposition and not to the reordered inline CSS-position.
If I have to set up the whole Elements again in JS ist a mess and not really a performance solution.

projecturl: [https://jsbin.com/firetuh/edit?html,css,js,output](https://jsbin.com/firetuh/edit?html,css,js,output)

@juliangarnier Can I ask you why it is not possible to get the targets CSS? If the Dom is loaded the CSS-Attributes of an element should be reading by javascript. And I do it a lot of times :).

Another problem of my animation, what I don't understand, is why the connector-elements not animate the height? Can this someone answer me?

It my first comprehensively project with anime.js and I have to say its really cool what I can do with it.
But I still have a lot to learn :).

Best, Jan
